### PR TITLE
Project API integration bug fix.

### LINKED ISSUE #2023: Project Selection Bug in litmus-portal.
**What happened**:

Project selection not responding.

**What you expected to happen**:

Project to be selected from dropdown.

**How to reproduce it (as minimally and precisely as possible)**:

Open project selection dropdown.

**Anything else we need to know?**:
Minor Bug in state management.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

Project selection bug fixes.

fixes #2023

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
High PriorityBug Fix PR for RC1

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@rajdas98 @Sanjay1611 @Saranya-jena PTAL.

File: litmus-portal/frontend/src/containers/layouts/Header/ProfileDropdownItems.tsx [hunk 1]
```diff
@@ -96,6 +106,10 @@ const ProfileInfoDropdownItems: React.FC<ProfileInfoDropdownItemProps> = ({
     setSwitchableProjects(projectsAvailableForSwitching);
   }, []);
 
+  useEffect(() => {
+    setSwitchableProjects(projects);
+  }, [loading]);
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Header/ProfileDropdownItems.tsx:111]: change the dependency to data not loading

File: litmus-portal/frontend/src/containers/layouts/Header/ProfileDropdownItems.tsx [hunk 2]
```diff
@@ -51,6 +55,12 @@ const ProfileInfoDropdownItems: React.FC<ProfileInfoDropdownItemProps> = ({
   const initials = nameSplit[1]
     ? userAvatar(name, false)
     : userAvatar(name, true);
+  // Query to get user details
+  const { data, loading } = useQuery<
+    CurrentUserDetails,
+    CurrentUserDedtailsVars
+  >(GET_USER, { variables: { username } });
+  const projects: Project[] = data?.getUser.projects ?? [];
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Header/ProfileDropdownItems.tsx:63]:
```suggestion
  const projects = data?.getUser.projects ?? [];
```
no need to type this....it's already inferred by the ts compiler

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee PTAL. cc: @rajdas98

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by rajdas98 — APPROVED** [review]

**Pull request merged**